spark_pipeline_framework/transformers/framework_file_downloader/v1/framework_file_downloader.py

```diff
@@ -0,0 +1,77 @@
+from typing import Optional, Dict, Any, List
+
+from pyspark import keyword_only
+from pyspark.ml.param import Param
+from pyspark.sql import DataFrame
+
+from spark_pipeline_framework.progress_logger.progress_log_metric import (
+    ProgressLogMetric,
+)
+from spark_pipeline_framework.progress_logger.progress_logger import ProgressLogger
+from spark_pipeline_framework.transformers.framework_file_transformer.v1.framework_file_transformer import (
+    FrameworkFileTransformer,
+)
+from spark_pipeline_framework.utilities.file_downloader import FileDownloader
+
+
+class FrameworkFileDownloader(FrameworkFileTransformer):
+    # noinspection PyUnusedLocal
+    @keyword_only
+    def __init__(
+        self,
+        download_urls: List[str] = None,
+        download_to_path: Optional[str] = "./",
+        name: Optional[str] = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        progress_logger: Optional[ProgressLogger] = None,
+        extract_zips: bool = False,
```

Comment:
can we make this `extract_zips` variable Optional as well?
cos the file type we're trying to download may not be in an "archive" file format.